pkg/apis/datasource/v0alpha1/config.go

```diff
@@ -0,0 +1,76 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type GenericDataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// generic config
+	Spec GenericDataSourceSpec `json:"spec"`
+
+	// Secure values placeholder (true for fields that exist)
+	Secure map[string]SecureValue `json:"secure,omitempty"`
+}
+
+type SecureValue struct {
+	// The input is only valid for writing the value -- it is replaced on read
+	Input string `json:"input,omitempty"`
```

Comment:
This is the type I mentioned to you that we could make it "common": https://github.com/grafana/grafana/blob/main/pkg/apis/secret/v0alpha1/exposed_secure_value.go#L14
  
It is quite pedantic in that you can only get the data out once, but since this input here will handle raw secrets it might be useful.